databases/week3/main.sql

```diff
@@ -0,0 +1,254 @@
+    `description`,
+    `location`,
+    `when`,
+    max_reservations,
+    price,
+    created_date
+  )
+VALUES
+  (
+    "4",
+    "pork sandwich",
+    "BBQ pulled pork sandwich with baked beans",
+    "Århus",
+    "2022-03-25 012:45:05",
+    "10",
+    "50",
+    "2022-03-25"
+  );
+--3.Get a meal with any id, fx 1
+SELECT
+  *
+FROM
+  meal
+WHERE
+  meal.id = 2;
+--4.Update a meal with any id, fx 1.
+  --Update any attribute fx the title or multiple attributes
+UPDATE
+  meal
+SET
+  max_reservations = "20",
+  price = "95"
+WHERE
+  meal.id = 2;
+--5.Delete a meal with any id
+DELETE FROM
+  meal
+WHERE
+  id = 3;
+--Reservation
+  --1.Get all reservations
+SELECT
+  *
+FROM
+  reservation;
+--2.Add a new reservation
+INSERT INTO
+  reservation (
+    id,
+    number_of_guests,
+    meal_id,
+    created_date,
+    contact_phonenumber,
+    contact_name,
+    contact_email
+  )
+VALUES
+  (
+    "3",
+    "3",
+    "2",
+    "2022-03-27",
+    "+45-12345678",
+    "R. Ghosh Sarkar",
+    "rghosh@gmail.com"
+  );
+--3.Get a reservation with any id, fx 1
+SELECT
+  *
+FROM
+  reservation
+WHERE
+  reservation.id = 1;
+--4.Update a reservation with any id, fx 1.
+  --Update any attribute fx the title or multiple attributes
+UPDATE
+  reservation
+SET
+  created_date = "2022-03-26",
+  number_of_guests = "6",
+  meal_id = "2"
+WHERE
+  reservation.id = 2;
+--5.Delete a reservation with any id, fx 1
+DELETE FROM
+  reservation
+WHERE
+  reservation.id = "3";
+--Review
+  --1.Get all reviews
+SELECT
+  *
+FROM
+  review;
+--2.Add a new review
+INSERT INTO
+  review (
+    id,
+    title,
+    `description`,
+    meal_id,
+    stars,
+    created_date
+  )
+VALUES
+  (
+    "6",
+    "grilled chicken",
+    "good",
+    "1",
+    "4",
+    "2022.03.29"
+  );
+--3.Get a review with any id, fx 1
+SELECT
+  *
+FROM
+  review
+WHERE
+  review.id = 2;
+--4.Update a review with any id, fx 1.
+  --Update any attribute fx the title or multiple attributes
+UPDATE
+  review
+SET
+  review.title = "grilled chicken",
+  review.description = "great food!!!",
+  review.stars = "4"
+WHERE
+  review.id = 2;
+--5.Delete a review with any id, fx 1
+DELETE FROM
+  review
+WHERE
+  review.id = "3";
+--Additional queries
+  --1.Get meals that has a price smaller than a specific price fx 90
+SELECT
+  *
+FROM
+  meal
+WHERE
+  meal.price > 80;
+--2.Get meals that still has available reservations
+SELECT
+  meal.id "meal id",
+  meal.title "meal title",
+  reservation.number_of_guests "reservation number"
+FROM
+  meal
+  INNER JOIN reservation ON meal.id = reservation.meal_id
+WHERE
+  meal.max_reservations > reservation.number_of_guests;
+--3.Get meals that partially match a title.
+  -- Rød grød med will match the meal with the
+  --title Rød grød med fløde
+SELECT
+  *
+FROM
+  meal
+WHERE
+  title LIKE '%chicken%';
+--4.Get meals that has been created between two dates
+SELECT
+  *
+FROM
+  meal
+WHERE
+  created_date BETWEEN '2022-03-23'
+  AND '2022-03-24';
```

Comment:
for readability and clarity condition should be in one liner.

Reply:
ok.